pb/terrarium/provider/provider.proto

```diff
@@ -72,4 +78,27 @@ message EndProviderRequest {
     PUBLISH = 1;
   };
   Action action = 4;
-}
\ No newline at end of file
+}
+
+message ProviderRequest {
+  string name = 1;
+  string version = 2;
+  string os = 3;
+  string arch = 4;
+}
+
+message DownloadSourceZipRequest {
+  ProviderRequest provider = 1;
+}
+
+message SourceZipResponse {
+  bytes zip_data_chunk = 1;
+}
+
+message DownloadShasumRequest {
+  Provider provider = 1;
+}
+
+message DownloadShasumResponse {
+  bytes shasum_data_chunk = 1;
+}
```

Comment:
Move these to pb/terrarium/provider/services/storage.proto.

Reply:
Moved.